### ISSUE TITLE
Transport Rule code does not export rules to an XML

### ISSUE THREAD

**Issue by mikevlcek0127** [MicrosoftDocs/OfficeDocs-Exchange#3000]
The code in the export script did not create the transportrules.xml and also did not generate an error.  I used the following Powershell commands to properly export and import the Transport Rules outside of using the script from the article.

$file = Export-TransportRuleCollection
Set-Content -Path "C:\EOP\Export\TransportRules.xml" -Value $file.FileData -Encoding Byte

[Byte[]]$Data = Get-Content -Path "C:\EOP\Export\TransportRules.xml" -Encoding Byte -ReadCount 0
Import-TransportRuleCollection -FileData $Data

Mike

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 004da0a8-07d5-54a0-3bba-3e281ce6db0c
* Version Independent ID: 0569fc92-4336-09ac-9413-688671d21efb
* Content: [Move domains & settings from one EOP organization to another](https://docs.microsoft.com/en-us/exchange/standalone-eop/move-domains-and-settings-between-eop-orgs)
* Content Source: [Exchange/ExchangeHybrid/standalone-eop/move-domains-and-settings-between-eop-orgs.md](https://github.com/MicrosoftDocs/OfficeDocs-Exchange/blob/public/Exchange/ExchangeHybrid/standalone-eop/move-domains-and-settings-between-eop-orgs.md)
* Product: **m365-security**
* Technology: **mdo**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
@mikevlcek0127, thanks. I see what happened.

Based on anther GitHub Issue, I recently scoured through all Exchange topics that were using Get-Content and -Encoding Byte as that -Encoding Byte syntax only works in PowerShell 5.1, and not it 7.x.

So, I replaced the transport rule commands in code:

Export (write):

```powershell
$file = Export-TransportRuleCollection
[System.IO.File]::WriteAllBytes('.TransportRules.xml', $file.FileData)
```

Import (read):

```powershell
Add-Content $outfile "`n$Data = [System.IO.File]::ReadAllBytes('.TransportRules.xml')"
```

However, `.TransportRules.xml` is basically a typo; it literally creates/reads a file named `.TransportRules.xml`.

Based on the original code, it should have been `.\TransportRules.xml`, which creates/reads the file named `TransportRules.xml` in the %UserProfile% folder by default.

I added the ' \ ' in both places. The individual code lines appear to read/write the XML file correctly now.

Comments?
